### ISSUE TITLE
QGIS crash when labelling layer

### ISSUE THREAD

**Issue by angelagonzalez** [qgis/QGIS#39463]
## User Feedback

 I was labeling a layer of vectorial data and Qgis crashed again and again

## Report Details

**Crash ID**: [19789aa7754dd71934e8fd554cb0fc48c6e29537](https://github.com/qgis/QGIS/search?q=19789aa7754dd71934e8fd554cb0fc48c6e29537&type=Issues)

**Stack Trace**
<pre>
QMetaObject::activate :
QgsTask::processSubTasksForCompletion qgstaskmanager.cpp:273
QThreadPoolPrivate::reset :
QThread::start :
BaseThreadInitThunk :
RtlUserThreadStart :
</pre>

**QGIS Info**
QGIS Version: 3.10.10-A Coru�a
QGIS code revision: 8b051b9a43
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.0.4
Running against GDAL: 3.0.4

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 6.1.7601

**Comment by angelagonzalez** [comment]
## User Feedback

Qgis crashed again when I was attempting to config the labels of a vectorial layer.

## Report Details

**Crash ID**: [19789aa7754dd71934e8fd554cb0fc48c6e29537](https://github.com/qgis/QGIS/search?q=19789aa7754dd71934e8fd554cb0fc48c6e29537&type=Issues)


**Stack Trace**
<pre>
QMetaObject::activate :
QgsTask::processSubTasksForCompletion qgstaskmanager.cpp:273
QThreadPoolPrivate::reset :
QThread::start :
BaseThreadInitThunk :
RtlUserThreadStart :
</pre>


**QGIS Info**
QGIS Version: 3.10.10-A Coru�a
QGIS code revision: 8b051b9a43
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.0.4
Running against GDAL: 3.0.4



**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 6.1.7601

**Comment by roya0045** [comment]
Update to current version and use a clean profile to see if it crashes.

**Comment by rduivenvoorde** [comment]
@angelagonzalez or provide some example data so we can try to reproduce it...

**Comment by angelagonzalez** [comment]
## User Feedback

Qgis suddlenly crashed when I was trying to execute the 'wordwrap' command in a layer's label.

## Report Details

**Crash ID**: [19789aa7754dd71934e8fd554cb0fc48c6e29537](https://github.com/qgis/QGIS/search?q=19789aa7754dd71934e8fd554cb0fc48c6e29537&type=Issues)


**Stack Trace**
<pre>
QMetaObject::activate :
QgsTask::processSubTasksForCompletion qgstaskmanager.cpp:273
QThreadPoolPrivate::reset :
QThread::start :
BaseThreadInitThunk :
RtlUserThreadStart :
</pre>


**QGIS Info**
QGIS Version: 3.10.10-A Coru�a
QGIS code revision: 8b051b9a43
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.0.4
Running against GDAL: 3.0.4



**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 6.1.7601

**Comment by rduivenvoorde** [comment]
@angelagonzalez with this information we cannot help you I think. Can you provide some real example data (shp or geopackage in zip), and provide the steps to reproduce this?
You write 'use wordwrap', do the attributes maybe contain 'strange' characters?
If you provide a small dataset we can try to reproduce, this should not happen, but can be something with strange data in a dataset.
Else: please try to update to latest version and check if it is still the case.

**Comment by Gustry** [comment]
@angelagonzalez 

> Update to current version and use a clean profile to see if it crashes.

You should try with a new QGIS profile.
https://docs.qgis.org/testing/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles

**Comment by angelagonzalez** [comment]
> 
> 
> @angelagonzalez with this information we cannot help you I think. Can you provide some real example data (shp or geopackage in zip), and provide the steps to reproduce this?
> You write 'use wordwrap', do the attributes maybe contain 'strange' characters?
> If you provide a small dataset we can try to reproduce, this should not happen, but can be something with strange data in a dataset.
> Else: please try to update to latest version and check if it is still the case.

Hi, Richard:
Thank you for your rapid response.
I'm in Spain and I have **3.10.10-A Coruña** version. In my country this is the latest stable version (see Qgis website: **_https://www.qgis.org/es/site/forusers/download.html_** ).
The point is that I downloaded four shapefiles of more than 1 gigabyte each one, and I loaded them at the same time in a .qgz file.
I know that maybe it's too heavy for an only file, but I really need to do it this way, because I work with flood risk maps and it's highly important to be able to compare areas among them, for a given place.
I admit that my pc is a bit old (year: 2010), so this maight be the problem. It's a AMD Athlon(tm) IIx2 260 Processor 3.21 GHz, with 4,00 GB of RAM, 64bits, and Windosw 7 Ultimate Service Pack 1 as operating system.
In contrast, the crashes didn't happen at the beginning. It's a new issue that I first notice two days ago.

The data layers that I am working with, are the following:
https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q10.zip
https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q50.zip
https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q100.zip
https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q500.zip

And the labels that I was configuring for them are these ones:
- concat(TIPO_ZONA, '\n',ZONA)
- concat(TIPO_ZONA, '\n',ZONA, '\n' ,CLAVE_EXPE, '\n' ,ORGANISMO, '\n' , ESTUDIO)
- concat(TIPO_ZONA, '\n',ZONA, '\n' ,CLAVE_EXPE, '\n' ,wordwrap(ORGANISMO, 30), '\n' , wordwrap(ESTUDIO,24))
- ZONA

In addition, in the same session I have loaded the following WMS services:
http://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_Demarcacion_Cuencas_Hidrograficas_Intracomunitarias?
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_cuencas_tramos_red_hidrografica?
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_localizacion_puntos_negros_inundacion?
http://www.ign.es/wms-inspire/pnoa-ma

As I said before, maybe it's too much for my computer... but the point is that the problem didn't happen before.

Thanks a lot in advance for your help.

Best regards,
Angela

**Comment by angelagonzalez** [comment]
> 
> 
> @angelagonzalez
> 
> > Update to current version and use a clean profile to see if it crashes.
> 
> You should try with a new QGIS profile.
> https://docs.qgis.org/testing/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles

Hi, Gustry:
Thanks for your recommendation. It's been very useful because I dindn't know anything about profiles.
I created a new clean profile and, although I noticed a much better performance and a higher speed at once, in the end Qgis crashed again when I was trying to edit the text of one of the shape layers.
This time the code of the error has been a bit different:
## Report Details

**Crash ID**: [c6cd991d32be432c113e83294a1161fdbc2f9561](https://github.com/qgis/QGIS/search?q=c6cd991d32be432c113e83294a1161fdbc2f9561&type=Issues)


**Stack Trace**
<pre>
QPropertyAnimation::targetObject :
QOpenGLWidget::context :
QObject::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:418
QCoreApplication::notifyInternal2 :
QCoreApplicationPrivate::sendPostedEvents :
qt_plugin_query_metadata :
QEventDispatcherWin32::processEvents :
UserCallWinProcCheckWow :
DispatchMessageWorker :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main :
BaseThreadInitThunk :
RtlUserThreadStart :
</pre>


**QGIS Info**
QGIS Version: 3.10.10-A Coru�a
QGIS code revision: 8b051b9a43
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.0.4
Running against GDAL: 3.0.4



**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 6.1.7601

Best regards,
Angela

**Comment by gioman** [comment]
> Qgis crashed again when I was trying to edit the text of one of the shape layers.

@angelagonzalez ok, but does it crash doing the same operation with labels that lead to this report?

**Comment by angelagonzalez** [comment]
> 
> 
> > Qgis crashed again when I was trying to edit the text of one of the shape layers.
> 
> @angelagonzalez ok, but does it crash doing the same operation with labels that lead to this report?

Yes, it always happens when I make changes on the labels of shape layers.

**Comment by angelagonzalez** [comment]
> 
> 
> > > Qgis crashed again when I was trying to edit the text of one of the shape layers.
> > 
> > 
> > @angelagonzalez ok, but does it crash doing the same operation with labels that lead to this report?
> 
> Yes, it always happens when I make changes on the labels of shape layers.

## User Feedback

This crash has happenned when I have selected a different shape layer on the scrolling menu of the labelling dialog box.
This time I haven't edited anything, and I was logged as a new clean profile.

Note 1: Yesterday I reinstalled Qgis 3.10.10 twice: one as the main user of my PC, and the other time as administrator.

Note 2: the emergent warning of Qgis ('Qgis unexpectedly ended') has crashed after a while too. It seems that the more I use Qgis, the more crashes I get.

Go to bed now, thanks to all you and see you later.



## Report Details

**Crash ID**: [be168d9a9e75f6ed6759e76bc7a8a06239a44887](https://github.com/qgis/QGIS/search?q=be168d9a9e75f6ed6759e76bc7a8a06239a44887&type=Issues)


**Stack Trace**
<pre>
QCoreApplication::postEvent :
qt_qFindChildren_helper :
QMetaObject::activate :
QgsTask::statusChanged moc_qgstaskmanager.cpp:213
QgsTask::processSubTasksForCompletion qgstaskmanager.cpp:272
QThreadPoolPrivate::reset :
QThread::start :
BaseThreadInitThunk :
RtlUserThreadStart :
</pre>


**QGIS Info**
QGIS Version: 3.10.10-A Coru�a
QGIS code revision: 8b051b9a43
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.0.4
Running against GDAL: 3.0.4



**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 6.1.7601

**Comment by rduivenvoorde** [comment]
@angelagonzalez thanks for all this info, this is what we need to be able to reproduce it. 
Now see IF we can, and then hopefully fix it.

About latest version: 3.10 is the latest LTR (= Long Term Release), but 3.14 is newer, and within a week or so there will be a new 3.16 available.... You could try that one next week (or so, not sure when 3.16 will be online available)

**Comment by rduivenvoorde** [comment]
@angelagonzalez are you working in a certain area? Because I loaded the 4 shape files, gave them the most complex label (with wordwrap), added some wms's but no crash.
Maybe you were working in an area where there was a special character in the label, that is why I am asking...
Did you have special labeling configurations? Can you maybe share a screenshot of those?

![Screenshot-20201021102855-1856x829](https://user-images.githubusercontent.com/731673/96694068-3e802e80-1388-11eb-8fdc-f811f605d188.png)

QGIS should not crash ( although Windows is not as stable as Debian, my OS of choice ;-) ).
To be sure we/you should try to get QGIS to crash on another computer?
If I have time I will try, but I'm rather busy at the moment.

But maybe most efficient is to wait for 3.16, and if you check this in that version?
Without crash (or crash info) we cannot fix it.

**Comment by gioman** [comment]
>  Without crash (or crash info) we cannot fix it.

@angelagonzalez @rduivenvoorde if possible please share your project (and necessary data).

**Comment by rduivenvoorde** [comment]
Download the 4 'big' zips @angelagonzalez wrote, and put them next to the qgz file in this zipped project file:

[qgisproject.zip](https://github.com/qgis/QGIS/files/5414564/qgisproject.zip)

Then you should see the same project as I have above.

Note that I see in my debug info a lot of:

Warning: GEOS exception: TopologyException: Input geom 0 is invalid: Self-intersection at or near point 158584.97279999964 4131439.3233292326 at 158584.97279999964 4131439.3233292326

So it also seems that the downloaded data has a lot of invalid geometries in it...
I think QGIS (and gdal) became more robust in this recently, so please:
- try to upgrade your QGIS
- clean up the shp files (like throwing away the part of spain you do not need, and maybe use some algo to clean the geometries?

**Comment by gioman** [comment]
> Download the 4 'big' zips @angelagonzalez wrote,

@rduivenvoorde @angelagonzalez this is what I get now trying to download

![Screenshot_20201021_102417](https://user-images.githubusercontent.com/1951107/96700727-ab46f900-1387-11eb-945b-97ed4eab6c60.png)

Can you link/attach them in any other way?

**Comment by rduivenvoorde** [comment]
@gioman yes, I've seen that too... but trying one of the other url's it worked, and after that it kept working and I downloaded all. Plz try again, and if all fails, let me know and I can put it on a webserver here.

**Comment by gioman** [comment]
> In addition, in the same session I have loaded the following WMS services:
> http://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx
> http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_Demarcacion_Cuencas_Hidrograficas_Intracomunitarias?
> http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_cuencas_tramos_red_hidrografica?
> http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_localizacion_puntos_negros_inundacion?
> http://www.ign.es/wms-inspire/pnoa-ma

@angelagonzalez to try replicate your project I would like to know what layers have you added from the above WMS services and in what CRS is configured your project.

**Comment by gioman** [comment]
@angelagonzalez also no crashes here, QGIS 3.10.10 on Linux. More details are needed (preferably your .qgs project).

**Comment by angelagonzalez** [comment]
> 
> 
> > Without crash (or crash info) we cannot fix it.
> 
> @angelagonzalez @rduivenvoorde if possible please share your project (and necessary data).

Hi everyone, I've just got home! Thanks you all for your help and for your kind dedication.

I attach my Qgis project here. Its name is: **Sesion_Inundabilidad.qgz**

Related layers are (in this order):
http://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_Demarcacion_Cuencas_Hidrograficas_Intracomunitarias?
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_cuencas_tramos_red_hidrografica?
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_localizacion_puntos_negros_inundacion?
LAMINAS_Q10_MUY_ALTA_20191008 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q10.zip
LAMINAS_Q50_ALTA_200617 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q50.zip
LAMINAS_Q100_MEDIA_20191008 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q100.zip
LAMINAS_Q500_BAJA_20191008 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q500.zip
http://www.ign.es/wms-inspire/pnoa-ma

The last labels I configured were:
- For LAMINAS_Q10_MUY_ALTA_20191008 :
concat(replace(concat('Z.I. T=',
substr(TIPO_ZONA,
 1+strpos(TIPO_ZONA, '(' ), 
9)),')',''), '\n',ZONA, '\n' ,CLAVE_EXPE,'\n' , wordwrap(ESTUDIO,30), '\n' ,wordwrap(ORGANISMO, 30))
- For LAMINAS_Q50_ALTA_200617
concat(replace(concat('Z.I. T=',
substr(TIPO_ZONA,
 1+strpos(TIPO_ZONA, '(' ), 
9)),')',''), '\n' , rio)
- For LAMINAS_Q100_MEDIA_20191008:
concat(replace(concat('Z.I. T=',
substr(TIPO_ZONA,
 1+strpos(TIPO_ZONA, '(' ), 
9)),')',''), '\n' , rio)
- For LAMINAS_Q500_BAJA_20191008:
concat(replace(concat('Z.I. T=',
substr(TIPO_ZONA,
 1+strpos(TIPO_ZONA, '(' ), 
9)),')',''), '\n',ZONA, '\n' ,CLAVE_EXPE,'\n' , wordwrap(ESTUDIO,30), '\n' ,wordwrap(ORGANISMO, 30))

I've been testing at home right now and Qgis didn't crashed any time. It's very odd since the previous days it was almost impossible to work with this file, because of the continuous crashes.

I've even forced fast changes of zoom and labelling in order to make a crash happen, but nothing. So, at this point I don't have any idea of what the problem was. Moreover, some of you have reported no crashes with the same set of data.

The exact moment when the crashes happenned were:
1- When I made changes on the formula of any of the labels.
2- When I tried to change the font or the location of the labels.
3- When I was working on labelling a given layer, and later I chose another layer from the scroll menu.
4- When I changed the zoom.

None of those bugs are happening right now. I don't know why.

Oppssss, in the end, the crash has happend, just after CLOSING the project and EXITING Qgis. This is the report:

## User Feedback

This crash has happen just when I've closed the project.

## Report Details

**Crash ID**: [25302eaa003ede04bac451514750b4a6819a9c66](https://github.com/qgis/QGIS/search?q=25302eaa003ede04bac451514750b4a6819a9c66&type=Issues)


**Stack Trace**
<pre>
QgsVectorLayer::~QgsVectorLayer qgsvectorlayer.cpp:217
QgsVectorLayer::`vector deleting destructor' :
QgsMapLayerStore::removeMapLayers qgsmaplayerstore.cpp:160
QgsMapLayerStore::removeMapLayers qgsmaplayerstore.cpp:121
QgsMapLayerStore::removeAllMapLayers qgsmaplayerstore.cpp:203
QgsProject::clear qgsproject.cpp:793
QgisApp::closeProject qgisapp.cpp:11965
QgisApp::fileExit qgisapp.cpp:5699
QWidget::event :
QgisApp::event qgisapp.cpp:1928
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:418
QCoreApplication::notifyInternal2 :
QWidgetPrivate::close_helper :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:418
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processCloseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
UserCallWinProcCheckWow :
DispatchMessageWorker :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main main.cpp:1606
BaseThreadInitThunk :
RtlUserThreadStart :
</pre>


**QGIS Info**
QGIS Version: 3.10.10-A Coru�a
QGIS code revision: 8b051b9a43
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.0.4
Running against GDAL: 3.0.4



**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 6.1.7601

**Comment by angelagonzalez** [comment]
> 
> 
> > Without crash (or crash info) we cannot fix it.
> 
> @angelagonzalez @rduivenvoorde if possible please share your project (and necessary data).

Hi everyone, I've just got home! Thanks you all for your help and for your kind dedication.

I attach my Qgis project here. Its name is: Sesion_Inundabilidad.qgz

Related layers are (in this order):
http://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_Demarcacion_Cuencas_Hidrograficas_Intracomunitarias?
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_cuencas_tramos_red_hidrografica?
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_localizacion_puntos_negros_inundacion?
LAMINAS_Q10_MUY_ALTA_20191008 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q10.zip
LAMINAS_Q50_ALTA_200617 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q50.zip
LAMINAS_Q100_MEDIA_20191008 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q100.zip
LAMINAS_Q500_BAJA_20191008 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q500.zip
http://www.ign.es/wms-inspire/pnoa-ma

The last labels I configured were:

    For LAMINAS_Q10_MUY_ALTA_20191008 :
    concat(replace(concat('Z.I. T=',
    substr(TIPO_ZONA,
    1+strpos(TIPO_ZONA, '(' ),
    9)),')',''), '\n',ZONA, '\n' ,CLAVE_EXPE,'\n' , wordwrap(ESTUDIO,30), '\n' ,wordwrap(ORGANISMO, 30))
    For LAMINAS_Q50_ALTA_200617
    concat(replace(concat('Z.I. T=',
    substr(TIPO_ZONA,
    1+strpos(TIPO_ZONA, '(' ),
    9)),')',''), '\n' , rio)
    For LAMINAS_Q100_MEDIA_20191008:
    concat(replace(concat('Z.I. T=',
    substr(TIPO_ZONA,
    1+strpos(TIPO_ZONA, '(' ),
    9)),')',''), '\n' , rio)
    For LAMINAS_Q500_BAJA_20191008:
    concat(replace(concat('Z.I. T=',
    substr(TIPO_ZONA,
    1+strpos(TIPO_ZONA, '(' ),
    9)),')',''), '\n',ZONA, '\n' ,CLAVE_EXPE,'\n' , wordwrap(ESTUDIO,30), '\n' ,wordwrap(ORGANISMO, 30))

I've been testing at home right now and Qgis didn't crashed any time. It's very odd since the previous days it was almost impossible to work with this file, because of the continuous crashes.

I've even forced fast changes of zoom and labelling in order to make a crash happen, but nothing. So, at this point I don't have any idea of what the problem was. Moreover, some of you have reported no crashes with the same set of data.

The exact moment when the crashes happenned were:
1- When I made changes on the formula of any of the labels.
2- When I tried to change the font or the location of the labels.
3- When I was working on labelling a given layer, and later I chose another layer from the scroll menu.
4- When I changed the zoom.

None of those bugs are happening right now. I don't know why.

Oppssss, in the end, the crash has happend, just after CLOSING the project and EXITING Qgis. This is the report:
User Feedback

This crash has happen just when I've closed the project.
Report Details

Crash ID: 25302eaa003ede04bac451514750b4a6819a9c66

Stack Trace

QgsVectorLayer::~QgsVectorLayer qgsvectorlayer.cpp:217
QgsVectorLayer::`vector deleting destructor' :
QgsMapLayerStore::removeMapLayers qgsmaplayerstore.cpp:160
QgsMapLayerStore::removeMapLayers qgsmaplayerstore.cpp:121
QgsMapLayerStore::removeAllMapLayers qgsmaplayerstore.cpp:203
QgsProject::clear qgsproject.cpp:793
QgisApp::closeProject qgisapp.cpp:11965
QgisApp::fileExit qgisapp.cpp:5699
QWidget::event :
QgisApp::event qgisapp.cpp:1928
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:418
QCoreApplication::notifyInternal2 :
QWidgetPrivate::close_helper :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:418
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processCloseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
UserCallWinProcCheckWow :
DispatchMessageWorker :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main main.cpp:1606
BaseThreadInitThunk :
RtlUserThreadStart :

QGIS Info
QGIS Version: 3.10.10-A Coru�a
QGIS code revision: 8b051b9
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.0.4
Running against GDAL: 3.0.4

System Info
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 6.1.7601
[Sesion_Inundabilidad.zip](https://github.com/qgis/QGIS/files/5417467/Sesion_Inundabilidad.zip)

**Comment by angelagonzalez** [comment]
> 
> 
> > Without crash (or crash info) we cannot fix it.
> 
> @angelagonzalez @rduivenvoorde if possible please share your project (and necessary data).

Hi everyone, I've just got home! Thanks you all for your help and for your kind dedication.

I attach my Qgis project here. Its name is: **Sesion_Inundabilidad.qgz** The area of study is Andalucia (South of Spain)

Related layers are (in this order):
http://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_Demarcacion_Cuencas_Hidrograficas_Intracomunitarias?
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_cuencas_tramos_red_hidrografica?
http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_localizacion_puntos_negros_inundacion?
LAMINAS_Q10_MUY_ALTA_20191008 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q10.zip
LAMINAS_Q50_ALTA_200617 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q50.zip
LAMINAS_Q100_MEDIA_20191008 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q100.zip
LAMINAS_Q500_BAJA_20191008 obtained from: https://www.miteco.gob.es/app/descargas/descargafichero.aspx?f=laminas-q500.zip
http://www.ign.es/wms-inspire/pnoa-ma

The last labels I configured were:

For LAMINAS_Q10_MUY_ALTA_20191008 :
concat(replace(concat('Z.I. T=',
substr(TIPO_ZONA,
1+strpos(TIPO_ZONA, '(' ),
9)),')',''), '\n',ZONA, '\n' ,CLAVE_EXPE,'\n' , wordwrap(ESTUDIO,30), '\n' ,wordwrap(ORGANISMO, 30))
For LAMINAS_Q50_ALTA_200617
concat(replace(concat('Z.I. T=',
substr(TIPO_ZONA,
1+strpos(TIPO_ZONA, '(' ),
9)),')',''), '\n' , rio)
For LAMINAS_Q100_MEDIA_20191008:
concat(replace(concat('Z.I. T=',
substr(TIPO_ZONA,
1+strpos(TIPO_ZONA, '(' ),
9)),')',''), '\n' , rio)
For LAMINAS_Q500_BAJA_20191008:
concat(replace(concat('Z.I. T=',
substr(TIPO_ZONA,
1+strpos(TIPO_ZONA, '(' ),
9)),')',''), '\n',ZONA, '\n' ,CLAVE_EXPE,'\n' , wordwrap(ESTUDIO,30), '\n' ,wordwrap(ORGANISMO, 30))

I've been testing at home right now and Qgis didn't crashed any time. It's very odd since the previous days it was almost impossible to work with this file, because of the continuous crashes.

I've even forced fast changes of zoom and labelling in order to make a crash happen, but nothing. So, at this point I don't have any idea of what the problem was. Moreover, some of you have reported no crashes with the same set of data.

The exact moment when the crashes happenned were:
1- When I made changes on the formula of any of the labels.
2- When I tried to change the font or the location of the labels.
3- When I was working on labelling a given layer, and later I chose another layer from the scroll menu.
4- When I changed the zoom.

None of those bugs are happening right now. I don't know why.

Oppssss, in the end, the crash has happend, just after CLOSING the project and EXITING Qgis. This is the report:
User Feedback

This crash has happen just when I've closed the project.
Report Details

Crash ID: 25302eaa003ede04bac451514750b4a6819a9c66

Stack Trace

QgsVectorLayer::~QgsVectorLayer qgsvectorlayer.cpp:217
QgsVectorLayer::`vector deleting destructor' :
QgsMapLayerStore::removeMapLayers qgsmaplayerstore.cpp:160
QgsMapLayerStore::removeMapLayers qgsmaplayerstore.cpp:121
QgsMapLayerStore::removeAllMapLayers qgsmaplayerstore.cpp:203
QgsProject::clear qgsproject.cpp:793
QgisApp::closeProject qgisapp.cpp:11965
QgisApp::fileExit qgisapp.cpp:5699
QWidget::event :
QgisApp::event qgisapp.cpp:1928
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:418
QCoreApplication::notifyInternal2 :
QWidgetPrivate::close_helper :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:418
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processCloseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
UserCallWinProcCheckWow :
DispatchMessageWorker :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main main.cpp:1606
BaseThreadInitThunk :
RtlUserThreadStart :

QGIS Info
QGIS Version: 3.10.10-A Coru�a
QGIS code revision: 8b051b9
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.0.4
Running against GDAL: 3.0.4

System Info
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 6.1.7601
[Sesion_Inundabilidad.zip](https://github.com/qgis/QGIS/files/5417487/Sesion_Inundabilidad.zip)
![Area_of_Study_Malaga-Spain](https://user-images.githubusercontent.com/6812954/96751476-85dcde00-13cd-11eb-9217-bf66a8f72c0d.jpg)

**Comment by angelagonzalez** [comment]
> 
> 
> Download the 4 'big' zips @angelagonzalez wrote, and put them next to the qgz file in this zipped project file:
> 
> [qgisproject.zip](https://github.com/qgis/QGIS/files/5414564/qgisproject.zip)
> 
> Then you should see the same project as I have above.
> 
> Note that I see in my debug info a lot of:
> 
> Warning: GEOS exception: TopologyException: Input geom 0 is invalid: Self-intersection at or near point 158584.97279999964 4131439.3233292326 at 158584.97279999964 4131439.3233292326
> 
> So it also seems that the downloaded data has a lot of invalid geometries in it...
> I think QGIS (and gdal) became more robust in this recently, so please:
> 
>     * try to upgrade your QGIS
> 
>     * clean up the shp files (like throwing away the part of spain you do not need, and maybe use some algo to clean the geometries?

Hi, thanks a lot for the recommendations.

I tried a reduced version of this project -including only the South part of the four shape files (Andalucía)- just this morning at work. But surprisingly, **_Qgis crashed even before the project was fully loaded_**!!

Al work, I have a similar PC to mine's. It's a double core Intel of 3,6(? aprox) Ghz and 8 MB RAM. The operating system is Windows 7 64bits too, but Qgis version there is 3.10.9 (instead of 3.10.10) because we are not allowed to make any changes on the software. It can only be done by the i.t. department.

After a while, I opened the main project **(Sesion_Inundabilidad.qgz)** at work, and I didn't experienced any crashes.

As technicians (we are civil engineers) we cannot make any changes on the source shape layers, because they are provided by the Goverment and we have to work with them exactly as they are. That's why strange characters as 'ñ' cannot be removed (maybe they could, but we are not allowed to do that).

Maybe, as you said... the solution will be to wait for the 3.16 version and beg the I.T. department for installing it on our machines.

I'll do so at home, just when I know 3.16 version has released.

Thank you very much for your help, to all of you, and we will continue working as is, and later with 3.16 version.

Best regards and thanks again,

Angela

**Comment by angelagonzalez** [comment]
> 
> 
> > In addition, in the same session I have loaded the following WMS services:
> > http://ovc.catastro.meh.es/Cartografia/WMS/ServidorWMS.aspx
> > http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_Demarcacion_Cuencas_Hidrograficas_Intracomunitarias?
> > http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_cuencas_tramos_red_hidrografica?
> > http://www.juntadeandalucia.es/medioambiente/mapwms/REDIAM_localizacion_puntos_negros_inundacion?
> > http://www.ign.es/wms-inspire/pnoa-ma
> 
> @angelagonzalez to try replicate your project I would like to know what layers have you added from the above WMS services and in what CRS is configured your project.

MY CRS is ETRS89 UTM zone 30N - E.P.S.G. 25830

**Comment by gioman** [comment]
> Oppssss, in the end, the crash has happend, just after CLOSING the project and EXITING Qgis. This is the report:
> User Feedback
> 
> This crash has happen just when I've closed the project.

@angelagonzalez I was able to re-create your project. Still no crashes here.

**Comment by angelagonzalez** [comment]
> 
> 
> > Oppssss, in the end, the crash has happend, just after CLOSING the project and EXITING Qgis. This is the report:
> > User Feedback
> > This crash has happen just when I've closed the project.
> 
> @angelagonzalez I was able to re-create your project. Still no crashes here.

Thank you very much, Giovanni.
Apparently the issue is due to my PCs' configurations or/and my Qgis (old) versions, without discarding the fact that both computers are really old.
Let's see how new 3.16 version works. Anyway, I finally was able to do my work despite of the crashes, since I kept on opening the project again and again until I managed to label all the layers. After this, I made some other operations such as comparing with legal limits of the rivers, looking into where we can built, etc etc, ao in the end, the objective is accomplished.

For me, we can already close this issue.

Regards and thanks a lot again,
Angela :)
